src/NuGet.Core/NuGet.Packaging/Strings.resx

```diff
@@ -701,4 +701,15 @@ Valid from:</comment>
   <data name="NoRepositoryCountersignature" xml:space="preserve">
     <value>Verification settings require a repository countersignature, but the package does not have a repository countersignature.</value>
   </data>
+  <data name="ExtractionError_PackageSignatureIntegrityFailure" xml:space="preserve">
+    <value>Package integrity check failed for package '{0}' version '{1}' from source '{2}'</value>
```

Comment:
should we add some meaning to the "integrity check" failure? Like telling the user the package might have been tampered or compromised? Basically this means we consider the package is not safe for the user to use, should we be more explicit about it?

Reply:
I dont think we can imply any meaning from integrity check failure. It could be an I/O problem or a stream issue too. Further, because of the following we should not need anything else -

1. `NU3008` is always an error and we do not place the package on the machine. User should not be able to use the package because of the restore.
2. `NU3008` is a hyperlink and if we want to add details, we should do it on the docs page.